### PR TITLE
Refactor location picker

### PR DESCRIPTION
**Pull request by tbergq**
Moving logic from `LocationPickerScreen` to a component `LocationPicker`

Our navigation screens should not contain this logic.

### CODE REVIEW

**Comment by mrtnzlml** [comment]
@tbergq Please rebase it on top of the current master. Thanks.

**Comment by tbergq** [comment]
@mrtnzlml rebased

File: app/hotels/src/allHotels/searchForm/locationPicker/__generated__/LocationPickerScreen_cities_Query.graphql.js [hunk 1]
```diff
@@ -0,0 +1,217 @@
+/**
```
Comment (mrtnzlml) [inline app/hotels/src/allHotels/searchForm/locationPicker/__generated__/LocationPickerScreen_cities_Query.graphql.js]: @tbergq Now we have two `LocationPickerScreen_cities_Query.graphql.js` files. It obviously doesn't work very well if you are only moving the files...

Comment (tbergq) [reply app/hotels/src/allHotels/searchForm/locationPicker/__generated__/LocationPickerScreen_cities_Query.graphql.js]: wtf, I specifically searched for this filename, vscode found only one 😖

Comment (tbergq) [reply app/hotels/src/allHotels/searchForm/locationPicker/__generated__/LocationPickerScreen_cities_Query.graphql.js]: @mrtnzlml  should be deleted now 🙏

**Pull request merged**